src/hotspot/cpu/riscv/templateTable_riscv_64.cpp

```diff
@@ -2267,15 +2267,15 @@ void TemplateTable::resolve_cache_and_index(int byte_no, Register Rcache, Regist
   __ lbu(Rscratch, Rcache, in_bytes(ConstantPoolCache::base_offset() + ConstantPoolCacheEntry::indices_offset()) + 7 - (byte_no + 1));
 #endif
   // Acquire by cmp-br-isync (see below).
-  __ li(R12_ARG2, (int)code);
-  __ beq(Rscratch, R12_ARG2, Lresolved);
+  __ addi(R11_ARG1, R0, (int) code);
```

Comment:
li will be translated to this simple instruction. You don't need to inline this, it looks much harder to read.